solr/core/src/java/org/apache/solr/filestore/NodeFileStore.java

```diff
@@ -76,135 +66,54 @@ public SolrJerseyResponse getFile(String path, Boolean sync, String getFrom, Boo
     final var response = instantiateJerseyResponse(SolrJerseyResponse.class);
 
     if (Boolean.TRUE.equals(sync)) {
-      try {
-        fileStore.syncToAllNodes(path);
-        return response;
-      } catch (IOException e) {
-        throw new SolrException(SolrException.ErrorCode.BAD_REQUEST, "Error getting file ", e);
-      }
+      ClusterFileStore.syncToAllNodes(fileStore, path);
```

Comment:
I've been a bit confused between NodeFileStore, ClusterFileStore, DistribFileStore ;-).

Reply:
Yeah, the name conflicts are unfortunate.

On the bright side - I think we can probably make "NodeFileStore" and "NodeFileStoreApi" go away as part of this PR.  (Because with these APIs changes, all of our filestore endpoints will be under the "/cluster/filestore" path).

That would just leave:

- `ClusterFileStoreApi` - annotated interface defining our filestore APIs
- `ClusterFileStore` - implementations of those APIs
- `DistribFileStore` - internal implementation of various filestore operations (syncing, deleting, creating, etc.)

Reply:
Alright, the most recent iteration removes NodeFileStore/NodeFileStoreApi.  There's still some room for confusion between ClusterFileStoreApi, ClusterFileStore, and DistribFileStore, but overall this PR leaves things better than it found it in that regard 👍